osx/KA-Lite/KA-Lite/AppDelegate.m

```diff
@@ -43,58 +43,61 @@ - (BOOL)applicationShouldHandleReopen:(NSApplication *)theApplication hasVisible
 - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
     // Insert code here to initialize your application
     
-    // Setup the status menu item.
-    self.statusItem = [[NSStatusBar systemStatusBar] statusItemWithLength:NSVariableStatusItemLength];
-    [self.statusItem setImage:[NSImage imageNamed:@"favicon"]];
-    [self.statusItem setMenu:self.statusMenu];
-    [self.statusItem setHighlightMode:YES];
-    [self.statusItem setToolTip:@"Click to show the KA Lite menu items."];
-    
-    [self.kaliteDataHelp setToolTip:@"This will set the KALITE_HOME environment variable to the selected KA Lite data location. \n \nClick the 'Apply' button to save your changes and click the 'Start KA Lite' button to use your new data location. \n \nNOTE: To use your existing KA Lite data, manually copy it to the selected KA Lite data location."];
-
-    // Set the default status.
-    self.status = statusCouldNotDetermineStatus;
-    [self getKaliteStatus];
-    
-    // We need to show preferences if the following does not exist:
-    // 1. database does not exist
-    // 2. symlinked `kalite` executable
-    // Apply button is clicked, we run `kalite manage setup`.
-    bool mustShowPreferences = false;
-    @try {
-        checkEnvVars();
-        NSString *database = getDatabasePath();
-        if (!pathExists(database)) {
-            NSLog(@"Database not found, must show preferences.");
-            mustShowPreferences = true;
-        } else {
-            NSLog([NSString stringWithFormat:@"FOUND database at %@!", database]);
-        }
-
-        NSString *kalite = getUsrBinKalite();
-        if (!pathExists(kalite)) {
-            NSLog(@"kalite executable is not found, must show preferences.");
-            mustShowPreferences = true;
-            [self showStatus:statusFailedToStart];
-            showNotification(@"Kalite executable is not found. You need to reinstall the KA Lite application.");
-            
-        } else {
+    NSString *kalite = getUsrBinKalite();
+    
+    if (!pathExists(kalite)) {
+        NSLog(@"kalite executable is not found.");
+        [self showStatus:statusFailedToStart];
+        alertDialog(@"Kalite executable is not found. You need to reinstall the KA Lite application.");
+        //The application must terminate if kalite executable is not found.
+        [[NSApplication sharedApplication] terminate:nil];
+    } else
+    
+    {
```

Comment:
You can actually just remove the `else` already because you are terminating the app anyway.